gwpy/signal/qtransform.py

```diff
@@ -377,7 +435,227 @@ def transform(self, fseries, norm=True, epoch=None):
+        if outseg is None:
+            outseg = self.energies[0].span
+        frequencies = self.plane.frequencies
+        dtype = self.energies[0].dtype
+        # build regular Spectrogram from peak-Q data by interpolating each
+        # (Q, frequency) `TimeSeries` to have the same time resolution
+        xout = numpy.arange(*outseg, step=tres)
+        nx = xout.size
+        ny = frequencies.size
+        out = Spectrogram(numpy.empty((nx, ny), dtype=dtype),
+                          t0=outseg[0], dt=tres, frequencies=frequencies)
+        # record Q in output
+        out.q = self.plane.q
+        # interpolate rows
+        for i, row in enumerate(self.energies):
+            xrow = numpy.arange(row.x0.value, (row.x0 + row.duration).value,
+                                row.dx.value)
+            interp = InterpolatedUnivariateSpline(xrow, row.value)
+            out[:, i] = interp(xout)
+        if fres is None:
+            return out
+        # interpolate the spectrogram to increase its frequency resolution
+        # --- this is done because Duncan doesn't like interpolated images
+        #     since they don't support log scaling
+        interp = interp2d(xout, frequencies, out.value.T, kind='cubic')
+        if not logf:
+            outfreq = numpy.arange(
+                self.plane.frange[0], self.plane.frange[1], fres)
+        else:
+            # using `~numpy.logspace` here to support numpy-1.7.1 for EPEL7,
+            # but numpy-1.12.0 introduced the function `~numpy.geomspace`
+            logfmin = numpy.log10(self.plane.frange[0])
+            logfmax = numpy.log10(self.plane.frange[1])
+            outfreq = numpy.logspace(logfmin, logfmax, fres)
+        new = type(out)(interp(xout, outfreq).T,
+                        t0=outseg[0], dt=tres, frequencies=outfreq)
+        new.q = self.plane.q
+        return new
+
+    def table(self, snrthresh=5.5):
+        """Represent this `QPlane` as an `EventTable`
+
+        Parameters
+        ----------
+        snrthresh : `float`, optional
+            lower inclusive threshold on individual tile SNR to keep in the
+            table, default: 5.5
+
+        Returns
+        -------
+        out : `~gwpy.table.EventTable`
+            a table of time-frequency tiles on this `QPlane`
+
+        Notes
+        -----
+        Only tiles with signal energy greater than or equal to
+        `snrthresh ** 2 / 2` will be stored in the output `EventTable`.
+        """
+        from ..table import EventTable
+        # get plane properties
+        freqs = self.plane.frequencies
+        bws = 2 * (freqs - self.plane.farray)
+        # collect table data as a recarray
+        names = ('time', 'frequency', 'duration', 'bandwidth', 'energy')
+        rec = numpy.recarray((0,), names=names, formats=['f8'] * len(names))
+        for f, bw, row in zip(freqs, bws, self.energies):
+            ind, = (row.value >= snrthresh ** 2 / 2).nonzero()
```

Comment:
@alurban, I'm concerned here that if `snrthresh` is given as an `int`, then `snrthresh ** 2 / 2` is entirely composed of integers and might not do what you think it does for python < 3: e.g.

```python
>>> 5 ** 2 / 2
12
>>> 5 ** 2 / 2.
12.5
```

To be safe, can you please add a float decimal point:

```suggestion
            ind, = (row.value >= snrthresh ** 2. / 2.).nonzero()
```